core/src/test/scala/unit/kafka/server/KafkaServerTest.scala

```diff
@@ -154,6 +155,96 @@ class KafkaServerTest extends QuorumTestHarness {
     server.shutdown()
   }
 
+  @Test
+  def testClusterWideDisablementOfTieredStorageWithEnabledTieredTopic(): Unit = {
+    val tsEnabledProps = TestUtils.createBrokerConfigs(1, zkConnect).head
```

Comment:
please add tests for kraft as well and for each case when broker is a controller, when broker is a KraftServer and when broker is a ZkServer. Asking because some of the initialization code path varies amongst them and we can't be sure that the config validation is always called correctly in the future as well.